src/terrain/terrain.js

```diff
@@ -931,15 +931,12 @@ export class Terrain extends Elevation {
             }
         }
 
-        const fadingOrTransitioning = id => {
+        const isTransitioning = id => {
             const layer = this._style._layers[id];
             const isHidden = layer.isHidden(this.painter.transform.zoom);
-            const crossFade = layer.getCrossfadeParameters();
-            const isFading = !!crossFade && crossFade.t !== 1;
-            const isTransitioning = layer.hasTransition();
-            return layer.type !== 'custom' && !isHidden && (isFading || isTransitioning);
```

Comment:
Removing this might have quite a beneficial impact on terrain render cache, as this would reduce the need to entirely re-render an entire tile to the offscreen FBO and instead keep it cached for more frames. It's not frequent to have that code path enabled in default styles so it might not show in our benchmark, but as long as you used at least one draped layer with patterns this could start showing fairly noticeable performance difference.

Reply:
@karimnaaji awesome! It might not show in the current default styles, but in the upcoming refreshed styles it might make a good difference.